### PR TITLE
Added TransferDetailsReview component

### PR DESCRIPTION
**Pull request by owillborn**
*Issue #:* /bcgov/entity#13905
readonlyreview

*Description of changes:*
- Created TransferDetailsReview component for transfer review and confirm screen
- Added tests for new component

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/tests/unit/MhrInformation.spec.ts [hunk 1]
```diff
@@ -394,4 +394,43 @@ describe('Mhr Information', () => {
...
+    wrapper.find('#btn-stacked-submit').trigger('click')
+    await Vue.nextTick()
+    await Vue.nextTick()
+
+    // renders TranseferDetailsReview
+    expect(wrapper.findComponent(TransferDetailsReview).exists()).toBeTruthy()
+    const mhrTransferDetailsReviewComponent = wrapper.findComponent(TransferDetailsReview)
+
+    // displaying correct declared value
+    expect(mhrTransferDetailsReviewComponent.find('#declared-value-display')).toBeTruthy()
```
Comment (dimak1) [inline ppr-ui/tests/unit/MhrInformation.spec.ts]: Should have expected value, ie: `.exists()` of `.visible()`.

File: ppr-ui/tests/unit/MhrInformation.spec.ts [hunk 2]
```diff
@@ -394,4 +394,43 @@ describe('Mhr Information', () => {
...
+    const currentDeclaredValue = mhrTransferDetailsReviewComponent.find('#declared-value-display')
+    expect(currentDeclaredValue.text()).toBe('$123456.00')
+
+    // autofilled consideration and displaying correct consideration value
+    expect(mhrTransferDetailsReviewComponent.find('#consideration-display')).toBeTruthy()
```
Comment (dimak1) [inline ppr-ui/tests/unit/MhrInformation.spec.ts]: Same as above.

**Review by doug-lovett — APPROVED** [review]
One more high 5.

File: ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    <v-row>
+      <v-col cols="3">
+        <label class="generic-label">Consideration</label>
+      </v-col>
+      <v-col cols="9" class="gray7" id="consideration-display">{{ consideration }}</v-col>
+    </v-row>
+    <v-row>
+      <v-col cols="3">
+        <label class="generic-label"
+          >Bill of Sale Date of <br />
```
Comment (dimak1) [inline ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue]: The formatting is hard to read.

File: ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    const { getTransferDetails } = useMhrInformation()
+
+    const formatDate = (inputDate: string) => {
+      return moment(inputDate).format('MMMM D, YYYY')
+    }
+
+    const localState = reactive({
+      declaredValue: getTransferDetails().declaredValue,
+      consideration: getTransferDetails().consideration,
+      transferDate: formatDate(getTransferDetails().transferDate),
```
Comment (dimak1) [inline ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue]: We have a util `date-helper` that we can use to display the right format.

File: ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    TransferDetails
+  },
+  props: {
+    isMhrTransfer: {
+      type: Boolean,
+      default: true
+    }
+  },
+  setup (props) {
+    const { getTransferDetails } = useMhrInformation()
```
Comment (dimak1) [inline ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue]: We can use `getTransferDetails` directly from the store, without `useMhrInformation()`

**Review by dimak1 — CHANGES_REQUESTED** [review]
Looks great and could use few improvements.

File: ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue (hunk 5 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue]: @dimak1 Would this be an import instead?

Comment (dimak1) [reply ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue]: Yea, import a `useGetters()` and then use any getters you need, eg. `getMhrTransferDeclaredValue`, `getMhrTransferConsideration`, etc. Similar to what you are doing in `useMhrInformation`, just do it here.

File: ppr-ui/tests/unit/MhrInformation.spec.ts (hunk 1 above)
Comment (owillborn) [reply ppr-ui/tests/unit/MhrInformation.spec.ts]: good catch

File: ppr-ui/tests/unit/MhrInformation.spec.ts (hunk 2 above)
Comment (owillborn) [reply ppr-ui/tests/unit/MhrInformation.spec.ts]: good catch

File: ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue (hunk 3 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue]: Got rid of <br/> and formatted nicer

File: ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue (hunk 4 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrTransfers/TransferDetailsReview.vue]: Changed so it now uses pre-existing convertDate function

**Review by dimak1 — APPROVED** [review]

**Comment by dimak1** [comment]
Nicely done! 👍🏻 @owillborn

**Review by cameron-eyds — APPROVED** [review]
Nice job!

**Pull request merged**